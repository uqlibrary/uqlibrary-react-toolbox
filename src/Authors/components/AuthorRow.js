import React, {Component} from 'react';
import {PropTypes} from 'prop-types';

import Divider from 'material-ui/Divider';
import SocialPerson from 'material-ui/svg-icons/social/person';
import RaisedButton from 'material-ui/RaisedButton';

// import './Authors.scss';

class AuthorRow extends Component {

    static propTypes = {
        name: PropTypes.string.isRequired,
        authorID: PropTypes.number.isRequired,
        removeAuthor: PropTypes.func.isRequired
    };

    constructor(props) {
        super(props);
    }

    remove = () => {
        this.props.removeAuthor(this.props.authorID);
    }

    render() {
        return (
            <div className="Author">
                <Divider />
                <div className="row">
                    <div className="flex authorLeftIcon"><SocialPerson /></div>
                    <div className="flex-100 authorName">{this.props.name}</div>
                    <div className="flex authorRightIcon"><RaisedButton label="Remove" onClick={this.remove} /></div>
                </div>
            </div>
        );
    }
}

export default AuthorRow;
