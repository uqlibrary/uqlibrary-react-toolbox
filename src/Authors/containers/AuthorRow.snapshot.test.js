jest.dontMock('../components/AuthorRow');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import AuthorRow from '../components/AuthorRow';

function setup() {
    return shallow(<AuthorRow authorID={1} name="Test Author" removeAuthor={jest.fn()} />);
}

describe('AuthorRow snapshots test', () => {
    it('renders an Author Row', () => {
        const wrapper = setup();
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
