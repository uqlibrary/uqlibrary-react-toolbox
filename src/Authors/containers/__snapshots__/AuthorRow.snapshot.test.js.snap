// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorRow snapshots test renders an Author Row 1`] = `
<div
  className="Author"
>
  <Divider
    inset={false}
  />
  <div
    className="row"
  >
    <div
      className="flex authorLeftIcon"
    >
      <SocialPerson />
    </div>
    <div
      className="flex-100 authorName"
    >
      Test Author
    </div>
    <div
      className="flex authorRightIcon"
    >
      <RaisedButton
        disabled={false}
        fullWidth={false}
        labelPosition="after"
        onClick={[Function]}
        primary={false}
        secondary={false}
      />
    </div>
  </div>
</div>
`;
